### ISSUE TITLE
Probably a ClassDeclaration is created wrapping a CtClass representing an interface

### ISSUE THREAD

**Issue by ftomassetti** [javaparser/javasymbolsolver#14]
```
Exception in thread "main" java.lang.IllegalArgumentException: 
                            javassist.CtClassType@2d6a9952[changed public abstract interface class javax.persistence.EntityManagerFactory 
                            fields= 
                            constructors= 
                            methods=javassist.CtMethod@fa76482f[public abstract createEntityManager ()Ljavax/persistence/EntityManager;], 
                                    javassist.CtMethod@622ca8b8[public abstract createEntityManager (Ljava/util/Map;)Ljavax/persistence/EntityManager;], 
                                    javassist.CtMethod@33a644f9[public abstract close ()V], javassist.CtMethod@ebf0b295[public abstract isOpen ()Z], 
                                    ]
    at me.tomassetti.symbolsolver.javassistmodel.JavassistClassDeclaration.<init>(JavassistClassDeclaration.java:57)
    at me.tomassetti.symbolsolver.resolution.typesolvers.JarTypeSolver.tryToSolveType(JarTypeSolver.java:112)
    at me.tomassetti.symbolsolver.resolution.typesolvers.CombinedTypeSolver.tryToSolveType(CombinedTypeSolver.java:40)
    at me.tomassetti.symbolsolver.javaparsermodel.contexts.CompilationUnitContext.solveType(CompilationUnitContext.java:125)
    at me.tomassetti.symbolsolver.javaparsermodel.declarations.JavaParserClassDeclaration.solveType(JavaParserClassDeclaration.java:333)
    at me.tomassetti.symbolsolver.javaparsermodel.contexts.ClassOrInterfaceDeclarationContext.solveType(ClassOrInterfaceDeclarationContext.java:113)
    at me.tomassetti.symbolsolver.javaparsermodel.JavaParserFacade.convertToUsage(JavaParserFacade.java:436)
    at me.tomassetti.symbolsolver.javaparsermodel.JavaParserFacade.convertToUsage(JavaParserFacade.java:428)
    at me.tomassetti.symbolsolver.javaparsermodel.JavaParserFacade.convert(JavaParserFacade.java:481)
    at me.tomassetti.symbolsolver.javaparsermodel.declarations.JavaParserSymbolDeclaration.getType(JavaParserSymbolDeclaration.java:131)
    at me.tomassetti.symbolsolver.model.resolution.Value.from(Value.java:22)
    at me.tomassetti.symbolsolver.javaparsermodel.contexts.AbstractJavaParserContext.solveWithAsValue(AbstractJavaParserContext.java:72)
    at me.tomassetti.symbolsolver.javaparsermodel.contexts.ClassOrInterfaceDeclarationContext.solveSymbolAsValue(ClassOrInterfaceDeclarationContext.java:77)
    at me.tomassetti.symbolsolver.javaparsermodel.contexts.ConstructorContext.solveSymbolAsValue(ConstructorContext.java:60)
    at me.tomassetti.symbolsolver.javaparsermodel.contexts.StatementContext.solveSymbolAsValue(StatementContext.java:66)
    at me.tomassetti.symbolsolver.javaparsermodel.contexts.StatementContext.solveSymbolAsValue(StatementContext.java:88)
    at me.tomassetti.symbolsolver.javaparsermodel.contexts.MethodCallExprContext.solveSymbolAsValue(MethodCallExprContext.java:197)
    at me.tomassetti.symbolsolver.resolution.SymbolSolver.solveSymbolAsValue(SymbolSolver.java:42)
    at me.tomassetti.symbolsolver.resolution.SymbolSolver.solveSymbolAsValue(SymbolSolver.java:47)
    at me.tomassetti.symbolsolver.javaparsermodel.JavaParserFacade.getTypeConcrete(JavaParserFacade.java:211)
    at me.tomassetti.symbolsolver.javaparsermodel.JavaParserFacade.getType(JavaParserFacade.java:136)
    at me.tomassetti.symbolsolver.javaparsermodel.JavaParserFacade.getType(JavaParserFacade.java:130)
    at me.tomassetti.symbolsolver.javaparsermodel.contexts.MethodCallExprContext.solveMethodAsUsage(MethodCallExprContext.java:153)
    at me.tomassetti.symbolsolver.javaparsermodel.JavaParserFacade.solveMethodAsUsage(JavaParserFacade.java:494)
    at es.ujaen.Searcher$MethodCallVisitor.visit(Searcher.java:151)
    at com.github.javaparser.ast.expr.MethodCallExpr.accept(MethodCallExpr.java:73)
    at com.github.javaparser.ast.visitor.VoidVisitorAdapter.visit(VoidVisitorAdapter.java:152)
    at com.github.javaparser.ast.expr.AssignExpr.accept(AssignExpr.java:76)
    at com.github.javaparser.ast.visitor.VoidVisitorAdapter.visit(VoidVisitorAdapter.java:407)
    at com.github.javaparser.ast.stmt.ExpressionStmt.accept(ExpressionStmt.java:53)
    at com.github.javaparser.ast.visitor.VoidVisitorAdapter.visit(VoidVisitorAdapter.java:169)
    at com.github.javaparser.ast.stmt.BlockStmt.accept(BlockStmt.java:58)
    at com.github.javaparser.ast.visitor.VoidVisitorAdapter.visit(VoidVisitorAdapter.java:297)
    at com.github.javaparser.ast.body.ConstructorDeclaration.accept(ConstructorDeclaration.java:95)
    at com.github.javaparser.ast.visitor.VoidVisitorAdapter.visit(VoidVisitorAdapter.java:231)
    at com.github.javaparser.ast.body.ClassOrInterfaceDeclaration.accept(ClassOrInterfaceDeclaration.java:86)
    at com.github.javaparser.ast.visitor.VoidVisitorAdapter.visit(VoidVisitorAdapter.java:260)
    at es.ujaen.Searcher.search(Searcher.java:70)
    at es.ujaen.Searcher.searchReferences(Searcher.java:35)
    at es.ujaen.Engine.run(Engine.java:23)
    at es.ujaen.Main.main(Main.java:16)
    at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
    at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
    at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke(Method.java:497)
    at com.intellij.rt.execution.application.AppMain.main(AppMain.java:140)
```

**Comment by TCke83** [comment]
Hi,

i seem to run into the same problem while resolving a type. Any idea for a fix or a workaround?

**Comment by ftomassetti** [comment]
The fix should be rather straightforward: if the CtClass is an interface we should not wrap it as a JavassistClassDeclaration but as a JavassistInterfaceDeclaration

**Comment by ftomassetti** [comment]
Closed by 6086c13092fdc88634d25587ba019d0e51c20f28
